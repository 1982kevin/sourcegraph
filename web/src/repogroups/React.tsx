import { RepogroupMetadata } from './types'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import * as React from 'react'
export const react: RepogroupMetadata = {
    title: 'React',
    name: 'react',
    url: '/react',
    description:
        "Explore React repositories on GitHub. The search examples show how to use React Hook useState with various data types.",
    examples: [
        {
            title: 'Find imports of useState with regex search',
            exampleQuery: <>import [^;]+useState[^;]+ from 'react'</>,
            rawQuery: "import [^;]+useState[^;]+ from 'react'",
            patternType: SearchPatternType.regexp,
        },
        {
            title: 'Examples of useState with an object as the input parameter',
            description: `useState takes a single argument and can accept a primitive, an array or an object.
            Syntax for usage is [state_value, function_to_update_state_value] = useState (initial_state_value).`,
            exampleQuery: <>{'useState({:[string]})'}</>,
            rawQuery: 'useState({:[string]}) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter',
            exampleQuery: <>useState(:[string])</>,
            rawQuery: 'useState(:[string]) count:1000',
            patternType: SearchPatternType.structural,
        },
        {
            title: 'Examples of useState with any JavaScript data type as the input parameter in only TypeScript files',
            exampleQuery: (
                <>
                    useState(:[string])
                    <span className="search-keyword">lang:</span>typescript
                </>
            ),
            rawQuery: 'useState(:[string]) count:1000 lang:typescript',
            patternType: SearchPatternType.structural,
        },
    ],
    homepageDescription: 'Learn React with code examples.',
    homepageIcon: 'https://cdn4.iconfinder.com/data/icons/logos-3/600/React.js_logo-512.png',
}
