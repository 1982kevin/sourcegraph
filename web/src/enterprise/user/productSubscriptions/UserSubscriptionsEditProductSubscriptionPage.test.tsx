import React from 'react'
import * as H from 'history'
import * as GQL from '../../../../../shared/src/graphql/schema'
import { UserSubscriptionsEditProductSubscriptionPage } from './UserSubscriptionsEditProductSubscriptionPage'
import { of } from 'rxjs'
import { MemoryRouter } from 'react-router'
import { mount } from 'enzyme'

jest.mock('mdi-react/ArrowLeftIcon', () => 'ArrowLeftIcon')

jest.mock('./ProductSubscriptionForm', () => ({
    ProductSubscriptionForm: 'ProductSubscriptionForm',
}))

jest.mock('../../../tracking/eventLogger', () => ({
    eventLogger: { logViewEvent: () => undefined },
}))

const history = H.createMemoryHistory()
const location = H.createLocation('/')

describe('UserSubscriptionsEditProductSubscriptionPage', () => {
    test('renders', () => {
        expect(
            mount(
                <MemoryRouter>
                    <UserSubscriptionsEditProductSubscriptionPage
                        user={{ id: 'u' }}
                        match={{ isExact: true, params: { subscriptionUUID: 's' }, path: '/p', url: '/p' }}
                        history={history}
                        location={location}
                        isLightTheme={true}
                        _queryProductSubscription={() =>
                            of({
                                id: 's',
                                name: 'L-123',
                                invoiceItem: {
                                    __typename: 'ProductSubscriptionInvoiceItem',
                                    userCount: 123,
                                    expiresAt: '2020-01-01',
                                    plan: { __typename: 'ProductPlan', billingPlanID: 'bp' } as GQL.IProductPlan,
                                },
                                url: '/s',
                            })
                        }
                    />
                </MemoryRouter>
            ).children()
        ).toMatchSnapshot()
    })
})
