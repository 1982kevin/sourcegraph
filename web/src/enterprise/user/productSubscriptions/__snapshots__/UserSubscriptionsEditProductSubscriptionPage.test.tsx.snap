// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSubscriptionsEditProductSubscriptionPage renders 1`] = `
<Router
  history="[History]"
>
  <UserSubscriptionsEditProductSubscriptionPage
    _queryProductSubscription={[Function]}
    history="[History]"
    isLightTheme={true}
    location="[Location path=/]"
    match={
      Object {
        "isExact": true,
        "params": Object {
          "subscriptionUUID": "s",
        },
        "path": "/p",
        "url": "/p",
      }
    }
    user={
      Object {
        "id": "u",
      }
    }
  >
    <div
      className="user-subscriptions-edit-product-subscription-page"
    >
      <PageTitle
        title="Edit subscription"
      />
      <Link
        className="btn btn-link btn-sm mb-3"
        to="/s"
      >
        <LinkAnchor
          className="btn btn-link btn-sm mb-3"
          href="/s"
          navigate={[Function]}
        >
          <a
            className="btn btn-link btn-sm mb-3"
            href="/s"
            onClick={[Function]}
          >
            <ArrowLeftIcon
              className="icon-inline"
            />
             Subscription
          </a>
        </LinkAnchor>
      </Link>
      <h2>
        Upgrade or change subscription 
        L-123
      </h2>
      <ProductSubscriptionForm
        accountID="u"
        afterPrimaryButton={
          <small
            className="form-text text-muted"
          >
            An upgraded license key will be available immediately.
          </small>
        }
        history="[History]"
        initialValue={
          Object {
            "billingPlanID": "bp",
            "userCount": 123,
          }
        }
        isLightTheme={true}
        onSubmit={[Function]}
        primaryButtonText="Upgrade subscription"
        subscriptionID="s"
      />
    </div>
  </UserSubscriptionsEditProductSubscriptionPage>
</Router>
`;
